basic_release.sh

```diff
@@ -3,22 +3,30 @@
 # Inputs
 # $1 -- Directory to release
 # $2 -- Version number
+# $3 -- Optional flag -docker
+# $4 -- package_name (required if -docker is used)
 
-if [ -z "${1+x}" ] && [ -z "${2+x}" ]
+if [ -z "${1+x}" ] || [ -z "${2+x}" ] || { [ "$3" = "-docker" ] && [ -z "${4+x}" ]; }
```

Comment:
can we make this `--docker` to make the syntax consistent with `argparse` (even though it is an arg for bash).  I am not sure if this is allowed or good practice, but the less cognitive overload for my brain the better.